nixos/platform/network.nix

```diff
@@ -81,7 +85,15 @@ let
         ((filter (a: fclib.isIp6 a.ip) encAddresses) ++
          (filter (a: fclib.isIp4 a.ip) encAddresses));
 
-   quoteLabel = replaceStrings ["/"] ["-"];
+  quoteLabel = replaceStrings ["/"] ["-"];
+
+  vrfName = iface: "vrf${iface.vlan}";
+  vrfTable = iface: let
+    # the routing tables 0, 253, 254, and 255 are reserved by the
+    # kernel. this base number is chosen as an arbitrary offset for
+    # numbering kernel routing tables for vrfs
+    tableBase = 1000;
```

Comment:
HMMM! This is interesting, we should check back with our vlan/vxlan/... numbering scheme and whether that can fit without shifting the values ...?

Reply:
That's not a bad idea actually. We definitely have space in the numbering scheme as it stands at present, so we could reserve those four IDs and then map the virtual network ID one-to-one to kernel tables.

Reply:
agreed. add assertions and update the docs.